lib/devices.go

```diff
@@ -64,16 +64,18 @@ func Stop() {
 // It takes a callback function to be run anytime there is a change in Devices
 func Init(onUpdateCallback func([]Device)) []Device {
 	pollingStop = make(chan bool)
-	logtime, initialDevices := Refresh()
+	go func() {
+		logtime, initialDevices := Refresh()
 
-	for initialDevices == nil {
-		time.Sleep(1 * time.Second)
-		addErrorLog("Failed to enumerate devices. Retrying...", logtime)
-		onUpdateCallback(nil)
-		logtime, initialDevices = Refresh()
-	}
+		for initialDevices == nil {
+			time.Sleep(1 * time.Second)
+			addErrorLog("Failed to enumerate devices. Retrying...", logtime, StateError)
+			onUpdateCallback(nil)
+			logtime, initialDevices = Refresh()
+		}
 
-	onUpdateCallback(initialDevices)
+		onUpdateCallback(initialDevices)
+	}()
 	go func() {
```

Comment:
Now you're starting two separate goroutines, of which the second one depends on the first having been run, so .. this is not good.